src/socket.cr

```diff
@@ -116,8 +116,16 @@ class Socket < IO
   # sock.bind 1234
   # ```
   def bind(port : Int)
-    Addrinfo.resolve("::", port, @family, @type, @protocol) do |addrinfo|
-      system_bind(addrinfo, "::#{port}") { |errno| errno }
+    if family.inet?
+      address = "0.0.0.0"
+      address_and_port = "0.0.0.0#{port}"
```

Comment:
Oh, shoot, that's a very important typo that I would have hoped the tests would caught them, but it seems it didn't.

Reply:
Oh okey, it seems that [`system_bind`](https://github.com/crystal-lang/crystal/blob/fff2d480dcc8693b5337e8e45a083abbda31a701/src/crystal/system/unix/socket.cr#L46-L50)  actually doesn't use `addrstr` when binding the port.

It seems that's only used for the exception, so it would have never been caught. 

I wonder if that parameter is really needed, because doing `grep "def system_bind"` shows me that no platform actually uses it.

Reply:
ok I pushed the fix and also changed the one of ipv6 according to the RFC. The string is only used in an exception is raised so it shouldn't cause any trouble.